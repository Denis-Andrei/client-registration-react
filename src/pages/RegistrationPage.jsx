import { Main, GridRow, GridCol} from 'govuk-react';
import React from 'react';

export default function RegistrationPage ({children}) {

    return(
        <Main>
            <GridRow>
                <GridCol data-testid="content" setWidth="two-thirds" >
                    {children}
                </GridCol>
            </GridRow>
        </Main>
    )
}
