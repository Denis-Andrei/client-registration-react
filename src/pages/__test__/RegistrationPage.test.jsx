import React from 'react';
import ReactDOM from 'react-dom';
import RegistrationPage from '../RegistrationPage';
import { render, cleanup } from '@testing-library/react'
import renderer from "react-test-renderer"


afterEach(cleanup);

it("RegistrationPage() renders withoud crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<RegistrationPage />, div)
})

it("renders RegistrationPage() correctly", () => {
    const str = "children"
    const { getByTestId }  = render(<RegistrationPage children={str}/>)
    expect(getByTestId('content')).toHaveTextContent("children")
})

it("matches snapshot", () => {
    const tree = renderer.create(<RegistrationPage />).toJSON();
    expect(tree).toMatchSnapshot();
})
