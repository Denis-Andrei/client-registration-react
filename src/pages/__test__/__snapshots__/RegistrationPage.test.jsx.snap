// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="src__OuterContainer-sc-1ppxqxg-0 kqJDjQ"
>
  <div
    className="src__InnerContainer-sc-1ppxqxg-1 bwOECn"
  >
    <div
      className="src__StyledRow-sc-117e6fv-0 feKIQa"
    >
      <div
        className="src__StyledColumn-sc-1d2lq4e-0 eYsGKG"
        data-testid="content"
      />
    </div>
  </div>
</div>
`;
